private/pkg/wasm/wasm.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2020-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package wasm provides a Wasm runtime for plugins.
+package wasm
+
+import (
+	"context"
+
+	"pluginrpc.com/pluginrpc"
+)
+
+// CompiledModule is a Wasm module ready to be run.
```

Comment:
It feels like this should be called `Runner`

Reply:
From this comment https://github.com/bufbuild/buf/pull/3349#discussion_r1784915200 did you mean its fine to leave as a `CompiledModule`? The `Runner` for Wasm modules is provided by `pluginrpcutil.NewWasmRunner`. The Wasm module is runnable but it's not a generic runner of plugins.

Reply:
I did mean that initially, but this *is* factually a runner, so this should probably be called `Runner` itself as well. Another option would be `RunCloser`, which would match ie `ReadCloser, WriteCloser` in being a union of `Runner` and `Closer`, but it's a little weird as this `Close` requires a `context.Context`.

I think with all that said though - I guess yes, you can keep it as `CompiledModule` in this case.